### PR TITLE
[IMP] crm: no need for revenues to get rainbows

### PR DESCRIPTION
**Pull request by ija-odoo**
in crm when there is no expected revenues is set on opportunity,
then the rainbowman is displayed for the "first won". which
is correct behavior. 
while it does make sense to ask for a
revenue so that we can measure if this is a record or not,
it is not necessary to congratulate the user for his first
win.
in this commit the rainbowman will display even the expected revenue
is not set on opportunity for the "first won". we have achieved this by
removing the expected_revenue from the specific condition for this case.

task-id: 2410443

### CODE REVIEW

File: addons/crm/models/crm_lead.py [hunk 1]
```diff
@@ -818,7 +818,22 @@ def get_rainbowman_message(self):
...
+                user_won_leads_count = self.search_count([
+                    ('type', '=', 'opportunity'),
+                    ('user_id', '=', self.user_id.id),
+                    ('probability', '=', 100),
+                    ('date_closed', '>=', date_utils.start_of(today, 'year')),
+                    ('date_closed', '<', date_utils.end_of(today, 'year')),
+                ])
+                if user_won_leads_count == 1:
+                    message = _('Go, go, go! Congrats for your first deal.')
+                return message
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_lead.py:835]: I bet you will have some issues if user_won_leads_count is not 1.

**Review by tde-banana-odoo — CHANGES_REQUESTED** [review]

File: addons/crm/models/crm_lead.py (hunk 1 above)
Comment (dja-odoo) [reply addons/crm/models/crm_lead.py:835]: Not sure what issues. Can you explain ?

Comment (tde-banana-odoo) [reply addons/crm/models/crm_lead.py:835]: Sorry, message is defined above. Strange construct actually I was a bit lost.

Maybe you could make some "sections" to ease understanding, like
```
if not user_id or not team_id:
    return False
if not expected_revenue:
     # compute stuff wo revenue
     if len(leads) == 1:
        message = ...
        return message
    return False
# compute stuff with revenue
message = ...
return message
```

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**